app/org/maproulette/session/dal/UserDAL.scala

```diff
@@ -473,26 +513,85 @@ class UserDAL @Inject() (override val db:Database,
     *
     * @param osmID The OSM ID of the user
     * @param projectId The id of the project to remove the user from
-    * @param groupType The type of group to add 1 - Admin, 2 - Write, 3 - Read
+    * @param groupType The type of group to remove -1 - all, 1 - Admin, 2 - Write, 3 - Read
     * @param user The user making the request
     * @param c
     */
   def removeUserFromProject(osmID:Long, projectId:Long, groupType:Int, user:User)(implicit c:Connection=null) : Unit = {
     this.permission.hasProjectAccess(this.projectDAL.retrieveById(projectId), user)
     implicit val osmKey = osmID
-    implicit val requestingUser = user
+    implicit val requestingUser = User.superUser
+
     userGroupDAL.clearUserCache(osmID)
     this.cacheManager.withUpdatingCache(Long => retrieveByOSMID) { cachedUser =>
       this.withMRTransaction { implicit c =>
+        if (groupType == -1) {
+          SQL"""DELETE FROM user_groups
+                WHERE osm_user_id = $osmKey AND group_id IN
+                  (SELECT id FROM groups WHERE project_id = $projectId)
+            """.executeUpdate()
+        } else {
+          SQL"""DELETE FROM user_groups
+                WHERE osm_user_id = $osmKey AND group_id =
+                  (SELECT id FROM groups WHERE group_type = $groupType AND project_id = $projectId)
+            """.executeUpdate()
+        }
+      }
+      Some(cachedUser.copy(groups = userGroupDAL.getUserGroups(osmID, User.superUser)))
+    }.get
+  }
+
+  /**
+    * Sets the group type of a user in a project, first deleting any prior group types,
+    * in a single transaction.
+    *
+    * @param osmID The OSM ID of the user
+    * @param projectId The id of the project to remove the user from
+    * @param groupType The type of group to set 1 - Admin, 2 - Write, 3 - Read
+    * @param user The user making the request
+    * @param c
+    */
+  def setUserProjectGroup(osmID:Long, projectId:Long, groupType:Int, user:User)(implicit c:Connection=null) : Unit = {
```

Comment:
I don't fully understand the use case around this function. Why would we need something to delete all user groups and then add them to just one. Is this like a sort of reset function?

Reply:
Ok I think this clears up things for me. For some reason I thought that the SQL statement was deleting the users association in other projects as well. I misread the SQL query. So I think this makes sense, as the permissions model is setup so that if you have Admin access you have write and read access, and if you have write access you have read access. So limiting to a single group type isn't an issue. I think this looks good, and thank you for the detailed explanation.